### PR TITLE
fix: Move OC_API into \OC\ApiHelper in standard namespace

### PR DESCRIPTION
**Pull request by come-nc**
## Summary

It’s only used by ocs/v1.php

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by come-nc** [comment]
I now see that v1.php only uses this in error handling, and that it’s kind of duplicated code with V1Response/V2Response.
But I’m not sure how hard it would be to switch over to using VXResponse classes instead.

**Comment by come-nc** [comment]
Integration test failure is related:

- [ ] In maintenance mode, even v1 returns 503 and not 200. (Not sure it makes sense but should be fixed for BC)

The taint errors I don’t know what to do with. The code path does not make sense, and in any case I’d expect json_encode output to always be safe to output. It’s only unsafe in a `<script>` tag and it’s not the case here.

File: lib/private/OCS/ApiHelper.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+use OCP\IRequest;
+use OCP\Server;
+
+class ApiHelper {
+	/**
+	 * Respond to a call
+	 * @psalm-taint-escape html
+	 * @param int $httpStatusCode force the HTTP status code, only used for the special case of maintenance mode which return 503 even for v1
+	 */
+	public static function respond(int $statusCode, string $statusMessage, array $headers = [], ?int $httpStatusCode = null): void {
```
Comment (artonge) [inline lib/private/OCS/ApiHelper.php:26]: Maybe more explicit?

```suggestion
	public static function respond(int $statusCode, string $statusMessage, array $headers = [], ?int $overrideStatusCode = null): void {
```

**Review by artonge — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]
https://github.com/nextcloud/server/blob/af6de04e9e141466dc229e444ff3f146f4a34765/build/psalm-baseline.xml#L2863-L2870 will be removed automatically?

**Comment by come-nc** [comment]
> https://github.com/nextcloud/server/blob/af6de04e9e141466dc229e444ff3f146f4a34765/build/psalm-baseline.xml#L2863-L2870
> 
> will be removed automatically?

Yes there’s a job for updating the baseline.

**Pull request merged**